libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.java

```diff
@@ -178,22 +180,20 @@ public boolean isValidRouteProfile(String routeProfile) {
   @Nullable
   public BannerInstructions findCurrentBannerInstructions(LegStep currentStep, double stepDistanceRemaining) {
     if (isValidStep(currentStep) && hasInstructions(currentStep.bannerInstructions())) {
-      int roundedDistanceRemaining = (int) stepDistanceRemaining;
       List<BannerInstructions> instructions = new ArrayList<>(currentStep.bannerInstructions());
-      Iterator<BannerInstructions> instructionsIterator = instructions.iterator();
-      while (instructionsIterator.hasNext()) {
-        BannerInstructions instruction = instructionsIterator.next();
-        double distanceAlongGeometry = instruction.distanceAlongGeometry();
-        if (distanceAlongGeometry < roundedDistanceRemaining) {
-          instructionsIterator.remove();
+      Collections.sort(instructions, new Comparator<BannerInstructions>() {
+        @Override
+        public int compare(BannerInstructions instructions1, BannerInstructions instructions2) {
+          return Double.compare(instructions1.distanceAlongGeometry(), instructions2.distanceAlongGeometry());
         }
+      });
+      for (BannerInstructions instruction : instructions) {
+        double distanceAlongGeometry = instruction.distanceAlongGeometry();
+          if (distanceAlongGeometry >= stepDistanceRemaining) {
+            return instruction;
+          }
       }
-      int instructionIndex = checkValidIndex(instructions);
-      if (instructions.size() > ZERO_INSTRUCTIONS) {
-        return instructions.get(instructionIndex);
-      }
-    }
-    return null;
+      return instructions.get(0);
   }
```

Comment:
I think we are missing a bracket here?  Also, it's fine to return `null` if the current instruction was not found, as the SDK is already built to handle this.

Reply:
if you return null (only on initialization), you can have empty instruction. So I prefer return the first one.

Reply:
@danesfeder is correct here. This implementation (as is right now) doesn't compile.